feat: 유저 관련 개선

pkdee opened the pull request:
> 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. <br>어떻게보다 무엇을 왜 수정했는지 설명해주세요.

### 🔗 관련 이슈
<!-- 연관된 이슈 번호를 적어주세요 -->
#14 #21 #27 #31
### ✨ 어떤 변경 사항이 있나요?
<!-- 실제 변경된 핵심 내용을 bullet로 작성해주세요 -->
- 유저 CRUD 구현
- state 무결성 검증
- 회원가입 로직 개선
- 랜덤 닉네임 API 구현

<!--
### ⚠️ 참고사항
_추가로 알아야 할 사항을 작성해주세요 (없다면 생략)_
- 현재는 A 조건에 대해서만 예외 처리를 했지만, 이후 B 조건도 고려 필요
- 응답 포맷 변경에 따라 프론트엔드 대응이 필요합니다
-->

Commit:
feat: JPA 설정 변경

- application.yml 파일에서 JPA 설정을 수정함
- 'ddl-auto' 옵션을 'update'에서 'create'로 변경

Commit:
#27 feat: 사용자 상태 및 MBTI 유형 변경

- UserEntity의 초기 상태를 PENDING으로 변경
- UserStatus에 PENDING 상태 추가
- UserProfileEntity의 MBTI 필드를 Enum 타입으로 변경
- UserMBTI Enum 클래스 추가, MBTI 16가지 유형 정의

Commit:
#31 feat: 사용자 프로필 및 닉네임 추천 기능 추가

- UserProfileDTO에 사용자 프로필 요청 및 닉네임 추천 응답 DTO 추가
- 닉네임 추천을 위한 NicknameService 인터페이스 추가
- NicknameService의 구현체인 NicknameServiceImpl 추가, 감정과 동물 이름 조합으로 랜덤 닉네임 생성 기능 구현

Commit:
#27 #31 feat: 사용자 프로필 생성 및 업데이트 로직 개선

- UserProfileDTO.Request를 UserProfileDTO.UserProfileRequest로 변경
- 사용자 상태가 PENDING인 경우에만 ACTIVE로 전환하는 조건 추가

Commit:
#21 #27 feat: 사용자 상태 정보 추가

- JwtTokenProvider에 사용자 상태 정보를 추가하여 AccessToken을 생성
- OAuth2SuccessHandler에서 JSON 응답에 사용자 상태 정보를 포함

Commit:
#21 feat: RefreshToken 재발급 로직 개선

- RefreshToken 검증 로직 강화
- Redis 업데이트 로직 추가
- 사용자 상태 정보 반환 로직 추가
- 로그 메시지 개선

Commit:
#21 #27 feat: 인증 서비스에 사용자 역할, 상태 추가

- CustomOidcUserService.kt 파일에 사용자 역할(UserRole), 상태(UserStatus)를 추가
- 사용자 저장 또는 갱신 시, 사용자 역할, 상태를 직접 전달하도록 변경
- 인증 객체 반환 시, 사용자 역할, 상태를 포함하여 반환하도록 수정

Commit:
#27 refactor: 사용자 인증 기능 개선

- OAuth 로그인 성공 시 사용자 등록 또는 로그인 시간 갱신 로직 개선
- 현재 로그인된 사용자 정보를 DTO로 반환하는 기능 추가
- 회원가입 시 프로필 저장 및 상태 ACTIVE로 전환하는 기능 추가
- RefreshToken을 검증하고 새로운 AccessToken + RefreshToken을 발급하는 기능 추가
- 로그아웃 요청 처리 (Request에서 AccessToken 추출) 기능 추가
- Redis에서 RefreshToken 제거 + AccessToken 블랙리스트 등록 로직 개선

Commit:
#21 #27 feat: 사용자 인증 정보 DTO 추가 및 토큰 DTO 수정

- 새로운 파일 AuthInfoDTO.kt 생성하여 사용자 인증 정보 응답 DTO 추가
- TokenDTO.kt에서 AccessToken 재발급 요청 DTO와 JWT 토큰 응답 DTO 수정
- JWT 토큰 응답 DTO에 사용자 상태 필드 추가

Commit:
#27 feat: 사용자 역할 및 상태 추가

- UserPrincipal 클래스에 UserRole 및 UserStatus 추가
- 사용자의 역할 및 상태 정보를 더 정확하게 관리하기 위함

Commit:
#27 #31 feat: AuthController 코드 리팩토링

- AuthController의 코드를 리팩토링하여 가독성과 유지보수성을 향상
- 불필요한 서비스와 메서드 제거, 필요한 서비스와 메서드 추가
- 각 메서드의 설명을 더 명확하게 변경
- '랜덤 닉네임 추천' 기능 추가

Commit:
#14 feat: 사용자 프로필 관련 코드 수정

- UserEntity의 withdrawnAt 필드를 var로 변경
- UserProfileDTO 클래스 설명 수정 및 필드 설명 추가
- UserProfileResponse 클래스 추가로 마이페이지 조회 기능 강화

Commit:
#14 feat: 사용자 프로필 관리 기능 추가

- UserProfileService에 프로필 생성 및 수정, 프로필 조회, 회원 탈퇴 기능 추가
- UserProfileServiceImpl에 각 기능의 구현 코드 추가
- 사용자 상태에 따른 프로필 생성 및 수정 로직 구현
- 회원 탈퇴 시 사용자 상태 변경 및 탈퇴 시간 기록 추가

Commit:
#14 feat: 사용자 컨트롤러 추가

- UserController.kt 파일 추가
- 사용자 프로필 조회, 수정, 회원 탈퇴 기능 구현
- 각 기능에 대한 API endpoint 설정
- 인증되지 않은 사용자에 대한 예외 처리 추가

Commit:
chore: [임시] CD 워크플로우에 auth-add 브랜치 추가

- main 브랜치 외에 auth-add 브랜치에서의 push도 CI/CD가 동작하도록 변경

Commit:
chore: [임시] GitHub Actions CD 설정 변경

- 'cd.yml' 파일의 push 이벤트 branch 설정을 'main', 'auth-add'에서 '*'로 변경

Commit:
chore: [임시] GitHub Actions CD 설정 변경

- 'cd.yml' 파일의 push 이벤트 branch 설정을 'main', 'auth-add'에서 '*'로 변경

Commit:
fix: GitHub Action 설정 변경

- main, dev 브랜치에 push시에만 deploy 작업이 실행되도록 cd.yml 파일 수정
- 이전에는 모든 브랜치에 대해 deploy 작업이 실행되었음

daersh reviewed (approved):
기능 확인 했습니다. 
### **_LGTM_**

Commit:
test: JWT 토큰에 유저 상태 정보 추가

- JwtTokenProvider에서 AccessToken 생성 시 유저 상태 정보를 포함하도록 변경
- AccessToken 파싱 시 유저 상태 정보를 추출할 수 있도록 수정
- 테스트 코드에 유저 상태 정보를 포함하여 토큰 생성 및 파싱 테스트 추가

9bfish8 reviewed (approved):
확인완완 고생하셨습니다🫡
